### ISSUE TITLE
Decide between `init`, lazy and passing values for startup values/singletons.

### ISSUE THREAD

**Issue by pbiggar** [darklang/dark#3944]
Lazy has the advantage that things startup faster, but the lazy things might fail. Lazy also allows us to do something after other dependencies (eg after telemetry is created), but it's not super good at that.

Init means we could forget to init something important.

Passing values moves the `init` to a weird place to ensure that the init phase is done.

We could do something like this:

- each value we want has an initialization function
- each value can register itself somehow
- we don't start up unless the all the initializers have been called

**Comment by StachuDotNet** [comment]
Folded into #5246
